### PR TITLE
Update for newer Elixir versions

### LINKED ISSUE #38: Upgrading Tesla
As requested @edgurgel 

I am having some difficulty upgrading Tesla to OTP 26 and the httparrot SSL config, check the CI status: https://github.com/elixir-tesla/tesla/pull/623

I also tried to use OTP 25+ built-in public key certs stuff, but it didn't work, I am pretty sure something in the configuration is broken that I am failing to see.

I would appreciate some help with it.

### PR DESCRIPTION
**Pull request by nathany-copia**
closes #38 

Compiles without warnings and all tests are passing in Elixir 1.16.

We'll have to see what Travis CI reports for older versions, and may need to adjust to support multiple versions at once. How many versions back would you like to support?

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -10,16 +11,19 @@ sudo: false
...
+      otp_release: '19.3'
+    - elixir: '1.7'
+      otp_release: '20.3'
+    - elixir: '1.7'
+      otp_release: '21.0'
+    - elixir: '1.8'
+      otp_release: '20.3'
+    - elixir: '1.8'
+      otp_release: '21.0'
+    - elixir: '1.8'
```
Comment (nathany-copia) [inline .travis.yml:24]: do you want to keep these old versions in the matrix? how far back?

File: mix.exs [hunk 2]
```diff
@@ -26,12 +26,12 @@ defmodule Httparrot.Mixfile do
...
     [
-      {:cowboy, "~> 2.8.0"},
+      {:cowboy, "~> 2.12.0"},
       {:exjsx, "~> 3.0 or ~> 4.0"},
-      {:con_cache, "~> 0.14.0"},
+      {:con_cache, "~> 1.1.0"},
       {:earmark, "~> 1.0", only: :dev},
       {:ex_doc, "~> 0.18", only: :dev},
-      {:meck, "~> 0.8.13", only: :test}
+      {:meck, "~> 0.9.2", only: :test}
```
Comment (nathany-copia) [inline mix.exs:34]: any reason to support older versions?

a lot of the deprecation warnings we've seen are coming from the old version of cowboy

Comment (nathany-copia) [reply mix.exs:34]: Also, let me know if you'd prefer to switch to GitHub Actions instead.

File: .travis.yml (hunk 1 above)
Comment (nathany-copia) [reply .travis.yml:24]: Also, let me know if you'd prefer to switch to GitHub Actions instead.

**Comment by nathany-copia** [comment]
This fixes the warnings and all the tests pass in Elixir 1.16, but it isn't enough to fix the test failures reported by Tesla.

File: .travis.yml (hunk 1 above)
Comment (edgurgel) [reply .travis.yml:24]: Hey nathany! thanks for starting this! Yes please I would love to move us to Github Actions!

File: mix.exs (hunk 2 above)
Comment (edgurgel) [reply mix.exs:34]: No reason to support old versions. We can bump everything!

File: .travis.yml (hunk 1 above)
Comment (edgurgel) [reply .travis.yml:24]: And happy to drop old versions. Happy to stay with the Elixir 1.15+ if that's possible

**Comment by edgurgel** [comment]
@nathany-copia I have a fix for the SSL issue once we get this merged :+1: Let me know if you want to get the GitHub Actions setup done. Cheers!

**Pull request closed without merging**